userland/src/sync_parameters.cpp

```diff
@@ -0,0 +1,96 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include <rclcpp/rclcpp.hpp>
+
+#include <simple_msgs/AllBuiltinTypes.h>
+#include <simple_msgs/AllDynamicArrayTypes.h>
+#include <simple_msgs/AllPrimitiveTypes.h>
+#include <simple_msgs/AllStaticArrayTypes.h>
+#include <simple_msgs/Nested.h>
+#include <simple_msgs/String.h>
+#include <simple_msgs/Uint32.h>
+
+#include <userland_msgs/AddTwoInts.h>
+
+int main(int argc, char ** argv)
+{
+  rclcpp::init(argc, argv);
+
+  auto node = rclcpp::Node::make_shared("parameters");
+
+  auto sync_parameters_client = std::make_shared<rclcpp::parameter::SyncParametersClient>(node);
+
+  auto parameters = {
+    rclcpp::parameter::ParameterVariant("foo", int64_t(2)),
+    rclcpp::parameter::ParameterVariant("bar", std::string("hello")),
+    rclcpp::parameter::ParameterVariant("baz", double(1.45)),
```

Comment:
Are the `int64_t` and `double` calls required, or is it just to be explicit in the example? It might be worth trying to expose the user to the idea that a literal will be stored as an `int64_t` or a `double` implicitly. We might do this by just passing the literal, but tagging on a comment which says that it will be stored as a `double` or whatever. I'm not sure what the most instructive style would be for the user.